types/datum.go

```diff
@@ -798,6 +798,9 @@ func (d *Datum) convertToString(sc *stmtctx.StatementContext, target *FieldType)
 		s = strconv.FormatFloat(d.GetFloat64(), 'f', -1, 64)
 	case KindString, KindBytes:
 		s = d.GetString()
+		if target.Tp == mysql.TypeString {
+			s=strings.TrimRight(s, " ")
```

Comment:
I am sure `TrimRight` is not the correct behavior.
```
mysql> insert into vctt values ("ab  ", "  ab   ");
mysql> select concat('(', v, ')'), concat('(', c, ')') from vctt;
+-------------------------------------------+
| concat('(', v, ')') | concat('(', c, ')') |
+-------------------------------------------+
| (ab  )              | (ab)                |
+-------------------------------------------+
3 row in set (0.002 sec)
```

Reply:
Please format the code before commit, thanks.

Reply:
```
TiDB:
mysql> select concat('(', v, ')'), concat('(', c, ')') from vctt;
+---------------------+---------------------+
| concat('(', v, ')') | concat('(', c, ')') |
+---------------------+---------------------+
| (ab  )              | (  ab)              |
| (ab  )              | (  ab)              |
+---------------------+---------------------+
2 rows in set (0.00 sec)

mysql:
mysql> select version();
+-----------+
| version() |
+-----------+
| 8.0.16    |
+-----------+
1 row in set (0.00 sec)

mysql> select concat('(', v, ')'), concat('(', c, ')') from vctt;
+---------------------+---------------------+
| concat('(', v, ')') | concat('(', c, ')') |
+---------------------+---------------------+
| (ab  )              | (  ab)              |
+---------------------+---------------------+
1 row in set (0.00 sec)

```

Reply:
I did not find specific trailing spaces in "https://dev.mysql.com/doc/refman/8.0/en/char.html".
I tried the following situations and the results are the same.
```
insert into vctt values ("ab  ", " a\a\a\a\a");
insert into vctt values ("ab  ", " a\b\b\b\b");
insert into vctt values ("ab  ", " a\f\f\f\f");
insert into vctt values ("ab  ", " a\v\v\v\v");

insert into vctt values ("ab  ", " a\n\n\n\n");
insert into vctt values ("ab  ", " a\r\r\r\r");
insert into vctt values ("ab  ", " a\t\t\t\t");
insert into vctt values ("ab  ", " a        ");

insert into vctt values ("ab  ", " a\\\\\\\\");
insert into vctt values ("ab  ", " a\'\'\'\'");
insert into vctt values ("ab  ", " a\"\"\"\"");
insert into vctt values ("ab  ", " a\?\?\?\?");
insert into vctt values ("ab  ", " a\0\0\0\0");
```